### ISSUE TITLE
Bitmap decoder issues

### ISSUE THREAD

**Issue by Kenny-Hui** [leezer3/OpenBVE#807]
### Description

Encountered this when playing with some older BVE content with the latest nightly build, some texture has gone very wrong
![image](https://user-images.githubusercontent.com/28094366/185390742-a3647b14-dcef-4e5c-9a20-2b7723a79e90.png)

1.8.3.2:
![image](https://user-images.githubusercontent.com/28094366/185391133-edc3b563-25a2-44a0-958b-3ef52b919901.png)

Probably something to do with the new Bitmap Decoder

### Reproduction

If the issue occurs in multiple routes/ trains, please provide one or two samples.
In order to reproduce the issue and debug it, it's helpful to have the following:

## Route
(There's another 7z inside, have to wrap it so github allows me to upload)
[MOL.zip](https://github.com/leezer3/OpenBVE/files/9373760/MOL.zip)

## Logs
[log.txt](https://github.com/leezer3/OpenBVE/files/9373779/log.txt)

### Related information 

* Operating system: Windows 10 21H2
* Method of control: Keyboard

**Comment by leezer3** [comment]
Definitely something to do with the new bitmap decoder.

Well technically, it's whatever encoded **kcrmosr(2009)\building\building01.bmp** failing to follow the BMP format specification.

**TLDR:**
The file in question declares that it has no color table (the color table length byte is zero), but yet it provides color data within the reserved color table block.
